default-enarx-trust-process.md

```diff
@@ -0,0 +1,474 @@
+   - provides the syscall processing from outside the Keep
+
+The initialisation of the process is prompted by the **Orchestrator**, which
+communicates with the **Enarx client agent**, which contacts the **Enarx
+host agent** over a network connection (which does not need to be
+encrypted, as no confidential data is passed across it, because the
+**Enarx host agent** is not a trusted entity - as it is part of the **tenant
+host domain** - anyway. A mechanism (such as a cryptographic nonce), allowing
+the Enarx client agent to match sessions and requests, is included in the
+initial request.  The **Enarx host agent** then requests the creation,
+which is performed by the **host CPU + CPU firmware**.
+
+The **Enarx runtime image** is typically stored on host storage (local or
+remote - the distinction is not relevant to the process).  The loading
+process varies between architectures, but is requested by the **Enarx
+host agent**.  Although the initial image may persist on host storage,
+there is no requirement for or against this, and so the component is not
+listed separately in state 1.  We follow this convention for later states.
+
+Once the **TEE instance** has been created, the **host CPU + CPU firmware**
+load the **Enarx runtime image** into the **TEE instance**.  The
+**Enarx runtime image**, now loaded, is executed, and becomes the
+**Enarx runtime instance**, though it is not yet attested.
+
+As part of the initialisation process, the **Orchestrator** provides the
+**Enarx client agent** with one of the following:
+ 1. access to the **tenant workload image repository** and a reference to
+ the  **tenant workload image** to be provisioned;
+ 2. direct access to the **tenant workload image**, either as a binary in
+ memory or in storage;
+ 3. a reference allowing the **Enarx client agent** to recontact the
+ **Orchestrator** in the event of successful attestation, in order then to
+ obtain the **tenant workload image**.
+ 
+Though the encryption of the channel between the Enarx client agent
+and the Enarx host agent is not **required**, the following properties are
+considered to be **desirable**:
+
+ - authorisation of the Enarx client agent to the Enarx host agent
+   - this allows the host to control which parties are permitted to create
+   Enarx Keeps.  While this has no direct benefit to the tenant, an
+   indirect benefit may be that the host can avoid DoS attacks, thereby
+   maintaining availability for workload placement and execution.
+   - note that the Enarx host agent is *not* trusted by the tenant, and
+   it is therefore the responsibility of the host to decide whether to
+   trust any such authorisation.
+ - integrity of the request
+   - some DoS attacks or protocol attacks may be feasible, and mitigated
+   by integrity protection.
+ - encryption of the protocol handshake and further steps
+   - some attacks based around protocol versioning or replays may be
+   feasible, and can be avoided by encryption.
```

Comment:
This part is not very convincing to me: in TLS, protocol versions are negotiated in plaintext and replays are protected by checking whether the message is tied to a specific handshake, based on integrity.

Reply:
We may not be using TLS for everything, and so I'm leaving open the possibility that any protocols might have weaknesses.  I'm not strongly attached to this text, however, so am happy to lose it if you prefer.